src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1013 @@
+
+    /* Retain a copy of the scene, also asserts `state` is in fact an `ufbx_scene` */
+    ufbx_retain(scene);
+
+    _state.reset(new State{});
+    _state->fromFile = !path.isEmpty();
+    _state->scene = ufbx_scene_ref{scene};
+
+    /* We need to split meshes into chunks per material, so precompute the
+       number of required chunks at the start, as eg. meshCount() depends on it */
+    {
+        UnsignedInt chunkCount = 0;
+        arrayResize(_state->meshChunkBase, (UnsignedInt)scene->meshes.count);
+
+        /* ufbx meshes can contain per-face materials so we need to separate them
+           into pieces containing a single material for SceneData. */
+        for(std::size_t i = 0; i < scene->meshes.count; ++i) {
+            ufbx_mesh *mesh = scene->meshes[i];
+
+            _state->meshChunkBase[i] = chunkCount;
+
+            for(const ufbx_mesh_material &mat : mesh->materials) {
+                if(mat.num_faces == 0) continue;
+                ++chunkCount;
+            }
+        }
+
+        arrayResize(_state->meshChunks, chunkCount);
+
+        /* Initialize mesh chunks */
+        for(ufbx_mesh *mesh : scene->meshes) {
+            UnsignedInt chunkOffset = _state->meshChunkBase[mesh->typed_id];
+            for (std::size_t i = 0; i < mesh->materials.count; ++i) {
+                const ufbx_mesh_material &mat = mesh->materials[i];
+                if(mat.num_faces == 0) continue;
+
+                MeshChunk &chunk = _state->meshChunks[chunkOffset];
+                chunk.meshId = mesh->typed_id;
+                chunk.meshMaterialIndex = (UnsignedInt)i;
+                ++chunkOffset;
+            }
+        }
+    }
+
+    /* Count the final number of nodes in the scene, we may remove some (root)
+       or add (synthetic geometry transform nodes) */
+    {
+        const bool preserveRootNode = configuration().value<bool>("preserveRootNode");
+        const bool geometricTransformNodes = configuration().value<bool>("geometricTransformNodes");
+
+        _state->nodeIdOffset = 0;
+        _state->originalNodeCount = (UnsignedInt)scene->nodes.count;
+
+        if(!preserveRootNode) {
+            --_state->originalNodeCount;
+            ++_state->nodeIdOffset;
+        }
+
+        _state->nodeCountWithSynthetic = _state->originalNodeCount;
+
+        /* Reserve space for nodes if we want to create dummy nodes for geometric
+           transforms */
+        if(geometricTransformNodes) {
+            for(ufbx_node *node : scene->nodes) {
+                if(node->has_geometry_transform) {
+                    ++_state->nodeCountWithSynthetic;
+                }
+            }
+        }
+    }
+}
+
+Int UfbxImporter::doDefaultScene() const { return 0; }
+
+UnsignedInt UfbxImporter::doSceneCount() const { return 1; }
+
+Int UfbxImporter::doSceneForName(Containers::StringView) {
+    return -1;
+}
+
+Containers::String UfbxImporter::doSceneName(UnsignedInt) {
+    return {};
+}
```

Comment:
If there's no scene names in FBX, you can just omit these two functions completely.